packages/beacon-node/src/network/core/mainThreadNetworkCore.ts

```diff
@@ -0,0 +1,37 @@
+import {NetworkCore} from "./types.js";
+import {BaseNetwork, BaseNetworkInit} from "./baseNetwork.js";
+
+async function init(modules: BaseNetworkInit): Promise<NetworkCore> {
+  const networkCore = {} as NetworkCore;
+
+  const base = await BaseNetwork.init(modules);
+
+  const reqResp = base.reqResp;
+  for (const [methodName, method] of Object.entries(reqResp)) {
+    if (method instanceof Function) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-call
+      (networkCore as any)[methodName] = method.bind(reqResp);
+    }
+  }
+  const gossip = base.gossip;
+  for (const [methodName, method] of Object.entries(gossip)) {
+    if (method instanceof Function) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-call
+      (networkCore as any)[methodName] = method.bind(gossip);
+    }
+  }
+
+  for (const [methodName, method] of Object.entries(base)) {
+    if (method instanceof Function) {
```

Comment:
I think it's best to have code repetition that is obvious than a smart solution like this to save a few lines. Verbose but repetitive patterns are very easy to maintainers at little extra cost if based on common primitives like fn that calls function

Reply:
Sounds good, I was going that route first, then tried to get clever :joy_cat:

Reply:
will do that tomorrow if you want to work on other stuff